### PR TITLE
Create a database for a clinic based on a diagram

### PR DESCRIPTION
**Pull request by bobb-Rob**
We implemented the following: 

- Created a file named schema_based_on_diagram.sql where we implement the database from the diagram.
- Created a many-to-many Join table  `treatments_medical_histories` for the relation between the `medical_histories` table and `treatments` table.

### CODE REVIEW

**Review by Ismailco — COMMENTED** [review]
Hi @bobb-Rob, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that should be created using peer programming

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: schema_based_on_diagram.sql [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+CREATE TABLE invoice_items (
+  id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+  unit_price DECIMAL,
+  quantity INT NOT NULL,
+  total_price DECIMAL,
+  invoice_id INT,
+  treatment_id INT,
+  FOREIGN KEY (treatment_id) REFERENCES treatments (id) ON DELETE CASCADE,
+  FOREIGN KEY (invoice_id) REFERENCES invoices (id) ON DELETE CASCADE
```
Comment (EricMbouwe) [inline schema_based_on_diagram.sql:46]: - [ ] all FOREIGN KEYS should have indexes created:

CREATE INDEX ON invoice_items (invoice_id);
CREATE INDEX ON invoice_items (treatment_id);

File: schema_based_on_diagram.sql [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  FOREIGN KEY (medical_history_id) REFERENCES medical_histories (id) ON DELETE CASCADE
+);
+
+CREATE TABLE invoices (
+  id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+  total_amount DECIMAL NOT NULL,
+  generated_at timestamp,
+  payed_at timestamp,
+  medical_history_id INT,
+  FOREIGN KEY (medical_history_id) REFERENCES medical_histories (id) ON DELETE CASCADE
```
Comment (EricMbouwe) [inline schema_based_on_diagram.sql:35]: - [ ] all FOREIGN KEYS should have indexes created:

CREATE INDEX ON invoices (medical_history_id);

File: schema_based_on_diagram.sql [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  type VARCHAR,
+  name VARCHAR NOT NULL
+);
+
+CREATE TABLE treatments_medical_histories (
+  id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+  treatment_id INT NOT NULL,  
+  medical_history_id INT NOT NULL,  
+  FOREIGN KEY (treatment_id) REFERENCES treatments (id) ON DELETE CASCADE,
+  FOREIGN KEY (medical_history_id) REFERENCES medical_histories (id) ON DELETE CASCADE
```
Comment (EricMbouwe) [inline schema_based_on_diagram.sql:26]: - [ ] all FOREIGN KEYS should have indexes created:

CREATE INDEX ON invoices (medical_history_id);
CREATE INDEX ON invoice_items (invoice_id);

File: schema_based_on_diagram.sql [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+ name VARCHAR(100),
+ date_of_birth DATE
+);
+
+CREATE TABLE medical_histories (
+  id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+  admitted_at timestamp NOT NULL,
+  patient_id INT, 
+  status VARCHAR NOT NULL,
+  FOREIGN KEY (patient_id) REFERENCES patients (id) ON DELETE CASCADE
```
Comment (EricMbouwe) [inline schema_based_on_diagram.sql:12]: - [ ] all FOREIGN KEYS should have indexes created:

CREATE INDEX ON medical_histories (patient_id);

**Review by EricMbouwe — CHANGES_REQUESTED** [review]
Hi @bobb-Rob, 

Good job so far 🔥!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 🔥🎉
- PR ✔
- Tables ✔

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ShadyShawkat — APPROVED** [review]
## Hi @bobb-Rob, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 
- Your readme looks professional ✔️ 

Image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**